mayastor/src/bdev/nexus/nexus_bdev_children.rs

```diff
@@ -138,10 +141,17 @@ impl Nexus {
                 // mark faulted so that it can never take part in the IO path of
                 // the nexus until brought online.
                 child.state = ChildState::Faulted;
+
                 self.children.push(child);
                 self.child_count += 1;
-                // TODO -- rsync labels
-                Ok(self.set_state(NexusState::Degraded))
+                self.set_state(NexusState::Degraded);
+
+                if let Err(e) = self.sync_labels().await {
+                    error!("Failed to sync labels {:?}", e);
+                    // todo: how to signal this?
```

Comment:
Can we create an issue for this? IMHO its rather important and not something we should shoehorn in later.

Reply:
@cjones1024 this is the one